scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Record.java

```diff
@@ -596,15 +598,22 @@ private Record(ProtocolVersion version, int epoch, long sequenceNumber, long rec
 				new IvParameterSpec(iv));
 		byte[] plaintext = blockCipher.doFinal(reader.readBytesLeft());
 		// last byte contains padding length
-		int paddingLength = plaintext[plaintext.length - 1];
+		int paddingLength = plaintext[plaintext.length - 1] & 0xff;
 		int fragmentLength = plaintext.length
```

Comment:
Just a detail but this is no more the fragment length ? (but I can find a better name)

Maybe we can keep :
```java
int fragmentLength = plaintext.length
				- 1 // paddingLength byte
				- paddingLength
				- currentReadState.getMacLength();
and check if 
``` java
if (fragmentLength < 0) {
	throw new InvalidMacException();
}
```

Reply:
If we agree to go for the  1 commit only, yes.